src/openrct2-ui/scripting/CustomWindow.cpp

```diff
@@ -1436,6 +1437,32 @@ namespace OpenRCT2::Ui::Windows
         }
     }
 
+    void CloseWindowsOwnedByPlugin(std::shared_ptr<Plugin> plugin)
+    {
+        // Get all the windows that need closing
+        std::vector<std::shared_ptr<rct_window>> customWindows;
+        for (const auto& window : g_window_list)
+        {
+            if (window->classification == WC_CUSTOM)
+            {
+                auto customWindow = reinterpret_cast<CustomWindow*>(window.get());
+                auto customInfo = reinterpret_cast<CustomWindowInfo*>(customWindow->custom_info);
+                if (customInfo != nullptr)
+                {
+                    if (customInfo->Owner == plugin)
+                    {
+                        customWindows.push_back(window);
+                    }
+                }
+            }
+        }
+
+        for (auto& window : customWindows)
+        {
+            window_close(window.get());
+        }
```

Comment:
Why not closing them as they are found? Is it because closing them already takes them off the `g_window_list`?

Reply:
Yes, it breaks the iterator.